### DISCUSSION TITLE
Long press to select objects on mobile / tablet / touch devices & simple press to pan

### DISCUSSION THREAD

**Discussion by kirill-konshin** [fabricjs/fabric.js#7602] [Show and tell]
I've put together a demo https://jsfiddle.net/dfuse/xhy5j9dp/ with desired functionality:

1. Press & drag = pan
2. Longpress & drag = selection

Demo works fine, but I am wondering if there's a more elegant solution?

In essence, the demo works like this:

1. I set up simple listeners for panning as in http://fabricjs.com/fabric-intro-part-5 and a timeout
2. If timeout hits, simple listeners are disposed
3. Create a rect, that represents selection (side question, is there a way to draw helper objects not on the main canvas?)
4. Dynamically adjust rect on mouse events
5. Once mouse is released I go through all objects and find if they intersect with rect http://fabricjs.com/intersection
6. Create new ActiveSelection with filtered objects, dispose rect

Additionally I've created a small visual clue for longpress. For sake of demo I'm not using true mobile handlers, just regular mouse handlers.

https://user-images.githubusercontent.com/1530410/149040678-600ee88d-9821-4d40-a1e3-910877afc75d.mp4

**Comment by asturur** [comment]
let me see if there is a nice solution.
your fiddle work and that is great already, maybe we can use internals a bit better to see if we can write less code.

**Reply by AlbertBadias** [reply]
Hi! any progressions on this?
I've try it on tablet but the panning mode makes the elements dissapear, and the long press it's a little bit tricky.

On pc works amazing btw :)

**Comment by ShaMan123** [comment]
You should use `groupSelector` logic I believe
